packages/browser-wallet/src/popup/shared/utils/hooks.ts

```diff
@@ -0,0 +1,74 @@
+import { Dispatch, SetStateAction, useEffect, useRef, useState } from 'react';
+
+export const useIsSubsequentRender = () => {
+    const ref = useRef<boolean>(false);
+
+    useEffect(() => {
+        ref.current = true;
+    }, [ref]);
+
+    return ref.current;
+};
+
+export const useUpdateEffect: typeof useEffect = (effect, deps) => {
+    const isSubsequentRender = useIsSubsequentRender();
+
+    useEffect(() => {
+        if (!isSubsequentRender) {
+            return undefined;
+        }
+        return effect();
+    }, deps);
+};
+
+/** Calls function at a given rate */
+export function useInterval(fn: () => void, rate: number, enable = true) {
+    useEffect(() => {
+        if (enable) {
+            const interval = setInterval(fn, rate);
+            return () => clearInterval(interval);
+        }
+        return () => {};
+    }, [enable, fn, rate]);
+}
+
+/** Hook for getting the current time, optionally taking a refresh frequency in
+ * milliseconds defaulting to 1 second */
+export function useCurrentTime(refreshRate = 1000) {
```

Comment:
This function and others in this file appear to be dead code. They should be removed and added if needed at a later point, but adding them now risks them just being dead forever.